components/style/values/computed/image.rs

```diff
@@ -54,6 +54,8 @@ pub enum LineDirection {
     Angle(Angle),
     /// A corner.
     Corner(X, Y),
+    /// A Position and an Angle for legacy `-moz-` prefixed gradient.
+    PositionAngle(Option<Position>, Option<Angle>),
```

Comment:
Because it can be position, angle or both. Position can be omitted when angle is specified or angle can be omitted when position is specified. But we check and return Errr in parsing code if both are None.